### PR TITLE
Rc sync action

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**
We have no automatic rc sync during releases.

**Description of the Change:**
Copy the PennyLane RC sync action and adapt it to Catalyst. It opens sync PR every morning during the feature freeze

### CODE REVIEW

File: .github/workflows/rc_sync.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      - name: Create temp branch
+        if: ${{ env.branch_exists == 'true' }}
+        run: |
+          git fetch
+          git checkout "${{ env.rc_branch }}"
+          BRANCH="rc_$(date +'%Y-%m-%d-%H-%M-%S')"
+          echo "tmp_branch=$BRANCH" >> $GITHUB_ENV
+          git checkout -b "$BRANCH"
+          # Exclude some files known to be divergent during the release process from the PR
+          git checkout main -- catalyst/frontend/_version.py
```
Comment (rmoyard) [inline .github/workflows/rc_sync.yaml:53]: @dime10 I am just not sure there will not be a lot a merge issue with the changelog as we have a single file compared to PL that splits it for each release.

Comment (dime10) [reply .github/workflows/rc_sync.yaml:53]: Hmm yeah the changelog might throw conflicts every time 🤔

Comment (dime10) [reply .github/workflows/rc_sync.yaml:53]: @erick-xanadu did you need to resolve conflicts in the changelog when you have done rc syncs to main?

Comment (erick-xanadu) [reply .github/workflows/rc_sync.yaml:53]: @dime10 I did need to resolve conflicts in the changelog.

Comment (rmoyard) [reply .github/workflows/rc_sync.yaml:53]: If we want to fix that we need to split the changelog into separate release files

Comment (rmoyard) [reply .github/workflows/rc_sync.yaml:53]: https://github.com/PennyLaneAI/catalyst/pull/1069 and https://github.com/PennyLaneAI/catalyst/pull/1070

**Comment by rmoyard** [comment]
[sc-72454]

File: .github/workflows/rc_sync.yaml (hunk 1 above)
Comment (rmoyard) [reply .github/workflows/rc_sync.yaml:53]: That should not be a blocker anymore

**Review by rauletorresc — APPROVED** [review]

**Pull request merged**